service/command.go

```diff
@@ -29,7 +31,17 @@ func NewCommand(set CollectorSettings) *cobra.Command {
 		RunE: func(cmd *cobra.Command, args []string) error {
 			featuregate.Apply(gatesList)
 			if set.ConfigProvider == nil {
-				set.ConfigProvider = MustNewDefaultConfigProvider(getConfigFlag(), getSetFlag())
+				var err error
+				cfgSet := newDefaultConfigProviderSettings()
+				cfgSet.Locations = getConfigFlag()
+				// Append the "overwrite properties converter" as the first converter.
+				cfgSet.MapConverters = append(
+					[]config.MapConverterFunc{overwritepropertiesmapconverter.New(getSetFlag())},
+					cfgSet.MapConverters...)
+				set.ConfigProvider, err = NewConfigProvider(cfgSet)
+				if err != nil {
+					return err
+				}
```

Comment:
This and the `collector_windows` code are the same, should we extract it to a function?